File: client/rental-service/src/app/user-info.ts
import {UserRole} from "./user-role";

export class UserInfo {
  email: string;
  id: number;
  name: string;
  lastName: string;
  role = UserRole.customer;

  public set(id: number, name: string, lastName: string, email: string, role: UserRole) {
    this.id = id;
    this.name = name;
    this.lastName = lastName;
    this.email = email;
    this.role = role;
  }

}

File: rest/src/main/java/com/nntu/containers/info/UserInfo.java
package com.nntu.containers.info;

import lombok.Data;

@Data
public class UserInfo {
    private Long id;
    private String name;
    private String lastName;
    private String email;
    private UserRole role;

    public UserInfo(Long id, String name, String lastName, String email, UserRole role) {
        this.id = id;
        this.name = name;
        this.lastName = lastName;
        this.email = email;
        this.role = role;
    }
}
